Libraries/LibURL/Origin.h

```diff
@@ -72,13 +75,16 @@ class Origin {
         // 2. If A and B are both tuple origins, run these substeps:
         if (!is_opaque() && !other.is_opaque()) {
             // 1. If A and B's schemes are identical, and their domains are identical and non-null, then return true.
-            // FIXME: Check domains once supported.
-            if (scheme() == other.scheme())
+            if (domain().has_value()
+                && other.domain().has_value()
+                && domain().value() == other.domain().value()
```

Comment:
These two lines can be reduced to: `domain() == other.domain()`

Reply:
Sorry I am a bit confused about the suggested change then, what are the two lines that can be simplified? I guess the .value() could be removed from both, but I suppose that was put there to avoid the == from checking .has_value() again